libminifi/src/core/ContentRepository.cpp

```diff
@@ -60,16 +60,29 @@ void ContentRepository::incrementStreamCount(const minifi::ResourceClaim &stream
   }
 }
 
+void ContentRepository::removeFromPurgeList() {
+  for (auto it = purge_list_.begin(); it != purge_list_.end();) {
+    if (removeKey(*it)) {
+      purge_list_.erase(it++);
+    } else {
+      ++it;
+    }
+  }
+}
+
 ContentRepository::StreamState ContentRepository::decrementStreamCount(const minifi::ResourceClaim &streamId) {
   std::lock_guard<std::mutex> lock(count_map_mutex_);
+  removeFromPurgeList();
   const std::string str = streamId.getContentFullPath();
   auto count = count_map_.find(str);
   if (count != count_map_.end() && count->second > 1) {
     count_map_[str] = count->second - 1;
     return StreamState::Alive;
   } else {
     count_map_.erase(str);
-    remove(streamId);
+    if (!remove(streamId)) {
+      purge_list_.push_back(streamId.getContentFullPath());
+    }
```

Comment:
why isn't this in `remove`?

Reply:
Updated in bde565c405e3e9419020445a48d41da561b58729